/**
 * Used to complete or fail an external task initiated through
 * {@link ExternalTask#doExecute(ExternalTaskCompletionHandle)}.
 * <p>
 * Flow framework is not thread safe and expects that all asynchronous code is
 * executed in a single thread. Currently ExternalTaskCompletionHandle is the
 * only exception as it allows {@link #complete()} and {@link #fail(Throwable)}
 * be called from other threads.
 * 
 * @author fateev
 */
